protocols/migration/migration_reports_6_0_0_to_reports_5_0_0.py

```diff
@@ -19,11 +18,174 @@ def migrate_interpretation_request_rd(self, old_instance):
         new_instance.versionControl = self.new_model.ReportVersionControl()
-        return self.validate_object(
-            object_to_validate=new_instance, object_type=self.new_model.InterpretationRequestRD
-        )
+        return self.validate_object(object_to_validate=new_instance, object_type=self.new_model.InterpretationRequestRD)
+
+    def migrate_interpreted_genome_to_interpreted_genome_rd(self, old_instance):
+        """
+        Migrates a reports_6_0_0.InterpretedGenome into a reports_5_0_0.InterpretedGenomeRD
+        :type old_instance: reports_6_0_0.InterpretedGenome
+        :rtype: reports_5_0_0.InterpretedGenomeRD
+        """
+        new_instance = self.convert_class(target_klass=self.new_model.InterpretedGenomeRD, instance=old_instance)
+        new_instance.versionControl = self.new_model.ReportVersionControl()
+        new_instance.variants = self.migrate_small_variants_to_reported_variants(small_variants=old_instance.variants)
+
+        return self.validate_object(object_to_validate=new_instance, object_type=self.new_model.InterpretedGenomeRD)
+
+    def migrate_small_variants_to_reported_variants(self, small_variants):
+        if small_variants is None:
+            msg = "Small Variants are required to migrate from InterpretedGenome version 6 to InterpretedGenomeRD "
+            msg += "version 5 as Reported Variants are required"
+            raise MigrationError(msg)
+        return [
+            self.migrate_small_variant_to_reported_variant(small_variant=small_variant)
+            for small_variant in small_variants
+        ]
+
+    def migrate_small_variant_to_reported_variant(self, small_variant):
+        """
+        Migrates a reports_6_0_0.SmallVariant into a reports_5_0_0.ReportedVariant
+        :type small_variant: reports_6_0_0.SmallVariant
+        :rtype: reports_5_0_0.ReportedVariant
+        """
+        new_instance = self.convert_class(target_klass=self.new_model.ReportedVariant, instance=small_variant)
+
+        var_attrs = small_variant.variantAttributes
+        if var_attrs:
+            new_instance.genomicChanges = var_attrs.genomicChanges
+            new_instance.cdnaChanges = var_attrs.cdnaChanges
+            new_instance.proteinChanges = var_attrs.proteinChanges
+            new_instance.additionalTextualVariantAnnotations = var_attrs.additionalTextualVariantAnnotations
+            new_instance.references = var_attrs.references
+            new_instance.additionalNumericVariantAnnotations = var_attrs.additionalNumericVariantAnnotations
+            new_instance.comments = var_attrs.comments
+            new_instance.alleleOrigins = var_attrs.alleleOrigins
+            new_instance.alleleFrequencies = [
+                self.convert_class(target_klass=self.new_model.AlleleFrequency, instance=allele_frequency)
+                for allele_frequency in var_attrs.alleleFrequencies
+            ]
```

Comment:
just realised this will fail because `alleleFrequencies` are nullable. But luckily they're nullable in both models so it's easy to fix ! :smile:

Reply:
PR #335 into this branch fixes this

Reply:
@priesgo I'll rebase #335 with this branch when you've rebased this branch with IP-1913